gui/hud.gui

```diff
@@ -3552,6 +3559,373 @@ types HUD
 		}
 	}
 
+	type progressbar_hud_infusion = progressbar_hud_stress {
+		size = { 100% 100% }
+
+		using = Animation_ShowHide_Standard
+	}
+
+	type infusion_icon = icon {
+		parentanchor = vcenter
+		size = { 30 30 }
+		framesize = { 70 70 }
+	
+		modify_texture = {
+			visible = "[Or(LessThanOrEqualTo_CFixedPoint(GetPlayer.MakeScope.Var('light_void_infusion_level').GetValue, '(CFixedPoint)-2'), GreaterThanOrEqualTo_CFixedPoint(GetPlayer.MakeScope.Var('light_void_infusion_level').GetValue, '(CFixedPoint)2'))]"
+			name = "lvl3_glow"
+			block "glow_color"
+			{
+				texture = "gfx/interface/colors/gold.dds"
+			}
+			blend_mode = colordodge
+			alpha = 0.2
+		}
+
+		modify_texture = {
+			name = "lvl_increase"
+			texture = "gfx/interface/colors/gold.dds"
+			blend_mode = colordodge
+			alpha = 0
+		}
+
+		state = {
+			name = "stress_level_increase"
+			next = "stress_level_increase_2"
+
+			modify_texture = {
+				name = "lvl_increase"
+				alpha = 0
+			}
+		}
+
+		state = {
+			name = "stress_level_increase_2"
+			next = "stress_level_increase_3"
+			duration = 0.4
+			bezier = { 0.5 0 1 0.5 }
+
+			modify_texture = {
+				name = "lvl_increase"
+				alpha = 0.7
+			}
+		}
+
+		state = {
+			name = "stress_level_increase_3"
+			duration = 0.5
+			bezier = { 0 0.5 0.5 1 }
+
+			modify_texture = {
+				name = "lvl_increase"
+				alpha = 0
+			}
+		}
+	}
+
+	#Warcraft
+	type infusion_bar = widget {
+		name = "infusion_widget"
+		size = { 170 55 }
+		#visible = "[Not(And(EqualTo_int32(FixedPointToInt(GetPlayer.MakeScope.Var('light_void_infusion_level').GetValue), '(int32)0'), EqualTo_int32(FixedPointToInt(GetPlayer.MakeScope.Var('light_void_infusion').GetValue), '(int32)0')))]"
+
+		widget = {
+			parentanchor = center
+			size = { 180 108 }
+
+			tooltip = "PLAYER_INFUSION_TOOLTIP"
+			using = tooltip_ne
+
+			widget = {
+				name = "progressbar"
+				parentanchor = center
+				widgetanchor = left|vcenter
+				position = { -18 0 }
+				size = { 80 20 }
+
+				background = {
+					texture = "gfx/interface/colors/white.dds"
+					color = { 0.1 0.1 0.1 0.8 }
+				}
+
+				hbox = { #Tried to make this into a type but it became longer
+					visible = "[GreaterThanOrEqualTo_int32(FixedPointToInt(GetPlayer.MakeScope.Var('light_void_infusion_level').GetValue), '(int32)1')]"
+					widget = {
+						layoutpolicy_horizontal = expanding
+						layoutpolicy_vertical = expanding
+						layoutstretchfactor_horizontal = "[GetProgressBarValueMaxScaled( Add_float(FixedPointToFloat(GetPlayer.MakeScope.Var('light_void_infusion').GetValue), '(float)100'), '(float)200', '(int32)80' )]"
```

Comment:
If I see it right, you do the math here. GUIs aren't good for this. You could create script values and refer them here. For example, you could have one value that regulates the infusion stretch.
https://ck3.paradoxwikis.com/Interface#Displaying_a_variable_or_script_value

Reply:
All of the bars now use values